controllers/metal3.io/host_state_machine.go

```diff
@@ -419,6 +419,14 @@ func (hsm *hostStateMachine) handleAvailable(info *reconcileInfo) actionResult {
 		return actionComplete{}
 	}
 
+	// Check if hostFirmwareSettings have changed
+	if dirty, _, err := hsm.Reconciler.getHostFirmwareSettings(info); err != nil {
```

Comment:
Wouldn't be  better to check it together with `getHostProvisioningSettings`? 

nit: just noticed that `getHostProvisioningSettings` is not part of `BareMetalHostReconciler`

Reply:
I think there needs to be a separate function for getHostFirmwareSettings() as its needed in actionPreparing() https://github.com/metal3-io/baremetal-operator/blob/main/controllers/metal3.io/baremetalhost_controller.go#L973>

Since in handleAvailable() its just a check to determine whether we move to Preparing I think its safe to have them separate, if we move to Preparing first because of getHostProvisioningSettings() that is OK.

Reply:
Ok, that was I noticed, the fact that the action could have been marked as completed prematurely